config/routes.rb

```diff
@@ -1,5 +1,9 @@
 Rails.application.routes.draw do
-  devise_for :users
+  devise_for :users, skip: [:registrations]
```

Comment:
I got this straight from the [Devise wiki on the subject](https://github.com/plataformatec/devise/wiki/How-To:-Allow-users-to-edit-their-password) so I would say it is probably ok, though I agree it does feel weird.